appinfo/application.php

```diff
@@ -92,23 +92,27 @@ public function __construct (array $urlParams = array()) {
 	 * Add wrapper for local storages
 	 */
 	public function setupWrapper(){
-		\OC\Files\Filesystem::addStorageWrapper('oc_avir', function ($mountPoint, $storage) {
-			/**
-			 * @var \OC\Files\Storage\Storage $storage
-			 */
-			if ($storage instanceof \OC\Files\Storage\Storage && $storage->isLocal()) {
-				$scannerFactory = $this->getContainer()->query('ScannerFactory');
-				$l10n = $this->getContainer()->query('L10N');
-				$logger = $this->getContainer()->query('Logger');
-				return new AvirWrapper([
-					'storage' => $storage,
-					'scannerFactory' => $scannerFactory,
-					'l10n' => $l10n,
-					'logger' => $logger
-				]);
-			} else {
-				return $storage;
-			}
-		});
+		\OC\Files\Filesystem::addStorageWrapper(
+			'oc_avir',
+			function ($mountPoint, $storage) {
+				/**
+				 * @var \OC\Files\Storage\Storage $storage
+				 */
+				if ($storage->instanceOfStorage('OC\Files\Storage\Local') && !$storage->instanceOfStorage('\OC\Files\Storage\MappedLocal')) {
```

Comment:
> It works when encryption is enabled because an antivirus wrapper is chained before an encryption wrapper. Which is proved by manual testing.

And what about external storages? They need to be virus scanned as well